File: PrintAllSubsequence/Subsequence.java
package PrintAllSubsequence;

class Subsequence{
    public static void main(String[] args) {
        String que = "abc";

        printAllSubsequence(que,"",0);
    }

    private static void printAllSubsequence(String que, String ans, int index) {
        if(index==que.length()){
            System.out.println(ans);
            return;
        }
        printAllSubsequence(que, ans, index+1);
        printAllSubsequence(que, ans+que.charAt(index), index+1);
    }
}

File: PrintAllSubsequence/subsequence.py

def printSubsequence(s,ans,index):
    if index >=len(s):
        print(ans)
        return
    
    printSubsequence(s,ans,index+1)
    printSubsequence(s,ans+s[index],index+1)
    



def main():
    s = "abc"

    printSubsequence(s,'',0)

main()